Lib/pydoc.py

```diff
@@ -389,8 +389,17 @@ def synopsis(filename, cache={}):
 class ErrorDuringImport(Exception):
     """Errors that occurred while trying to import something to document it."""
     def __init__(self, filename, exc_info):
+        if not isinstance(exc_info, tuple):
+            assert isinstance(exc_info, BaseException)
+            self.exc = type(exc_info)
+            self.value = exc_info
+            self.tb = exc_info.__traceback__
```

Comment:
self.tb is not used in this module. Would users expect it to be there?

Reply:
If there are users, then they might.  I think 'Its cheap, leave it.'  Not worth stressing over.

Reply:
More thought.  If stub file has class instance attributes (I am not sure), then *it* would expect it.  As a developer reading this, I would expect both branches to be consistent.

Reply:
I suggest changing the PR title to the positive "Let pydoc use the exception instead of sys.exc_info.  The PR does exactly this, without compatibility breaks.  Then merge.